src/commands/deploy/getDeployNode.ts

```diff
@@ -6,23 +6,24 @@
 import { Disposable, Uri } from "vscode";
 import { configurationSettings, none } from "../../constants";
 import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { TrialAppTreeItem } from '../../explorer/trialApp/TrialAppTreeItem';
 import { WebAppTreeItem } from "../../explorer/WebAppTreeItem";
 import { ext } from '../../extensionVariables';
 import { getWorkspaceSetting, updateWorkspaceSetting } from "../../vsCodeConfig/settings";
 import { IDeployContext, WebAppSource } from "./IDeployContext";
 
 export interface IDeployNode {
-    node: SiteTreeItem;
+    node: SiteTreeItem | TrialAppTreeItem;
     isNewWebApp: boolean;
 }
 
-export async function getDeployNode(context: IDeployContext, target: Uri | string | SiteTreeItem | undefined, isTargetNewWebApp: boolean): Promise<IDeployNode> {
+export async function getDeployNode(context: IDeployContext, target: Uri | string | SiteTreeItem | TrialAppTreeItem | undefined, isTargetNewWebApp: boolean): Promise<IDeployNode> {
     const defaultWebAppId: string | undefined = getWorkspaceSetting(configurationSettings.defaultWebAppToDeploy, context.workspace.uri.fsPath);
-    let node: SiteTreeItem | undefined;
+    let node: SiteTreeItem | TrialAppTreeItem | undefined;
 
     // if the entry point for deploy was via "Deploy" after creating, it is also a new web app (so confirmDeployment would be false)
     let isNewWebApp: boolean = isTargetNewWebApp;
-    if (target instanceof SiteTreeItem) {
+    if (target instanceof SiteTreeItem || target instanceof TrialAppTreeItem) {
```

Comment:
nit: You can just use `SiteTreeItemBase`.  It's cleaner, but adds another import, so iunno